pkg/bloombuild/planner/queue/queue_test.go

```diff
@@ -0,0 +1,210 @@
+package queue
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/services"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/v3/pkg/bloombuild/planner/plannertest"
+	"github.com/grafana/loki/v3/pkg/bloombuild/protos"
+	"github.com/grafana/loki/v3/pkg/storage"
+	v1 "github.com/grafana/loki/v3/pkg/storage/bloom/v1"
+	"github.com/grafana/loki/v3/pkg/storage/config"
+	"github.com/grafana/loki/v3/pkg/storage/stores/shipper/bloomshipper"
+)
+
+type taskMeta struct {
+	stat1 int
+	stat2 string
+}
+
+type taskWithMeta struct {
+	*protos.ProtoTask
+	*taskMeta
+}
+
+func createTasks(n int) []*taskWithMeta {
+	tasks := make([]*taskWithMeta, 0, n)
+	// Enqueue tasks
+	for i := 0; i < n; i++ {
+		task := &taskWithMeta{
+			ProtoTask: protos.NewTask(
+				config.NewDayTable(plannertest.TestDay, "fake"),
+				"fakeTenant",
+				v1.NewBounds(model.Fingerprint(i), model.Fingerprint(i+10)),
+				plannertest.TsdbID(1),
+				[]protos.Gap{
+					{
+						Bounds: v1.NewBounds(0, 10),
+						Series: plannertest.GenSeries(v1.NewBounds(0, 10)),
+						Blocks: []bloomshipper.BlockRef{
+							plannertest.GenBlockRef(0, 5),
+							plannertest.GenBlockRef(6, 10),
+						},
+					},
+				},
+			).ToProtoTask(),
+			taskMeta: &taskMeta{stat1: i, stat2: fmt.Sprintf("task-%d", i)},
+		}
+		tasks = append(tasks, task)
+	}
+	return tasks
+}
+
+func TestQueue(t *testing.T) {
+	for _, tc := range []struct {
+		name    string
+		useDisk bool
+	}{
+		{
+			name: "in-memory",
```

Comment:
nit: Could be more explicit:

```suggestion
			name:    "in-memory",
			useDisk: false,
```